include/cosim/algorithm/fixed_step_algorithm.hpp

```diff
@@ -30,8 +30,11 @@ class fixed_step_algorithm : public algorithm
      *
      *  \param baseStepSize
      *      The base communication interval length.
+     *
+     *  \param workerThreadCount
+     *      A number of worker threads to spawn for running FMUs
      */
-    explicit fixed_step_algorithm(duration baseStepSize);
+    explicit fixed_step_algorithm(duration baseStepSize, std::optional<unsigned int> workerThreadCount = std::nullopt);
```

Comment:
`optional` feels heavy-handed here. A more "C++-ish" solution would be to use a native integer type and plain zero instead of `nullopt`.

Reply:
`std::nullopt` means `std::hardware_currency-1`. 0 means 0. `std::optional` provides more information here.

Reply:
Ah, so the number refers to the number of _additional_ threads beyond the main thread? Then I completely agree.